[9.x] Added failOnFirstError property

yusufonur opened the pull request:
**Goal:**
During FormRequest validation, if a field fails, it will throw the validation error without checking the fields that come after it.

For example, we have defined rules as follows. In the present case; Even if the user_id field is missing, the post_id and comment_id fields are still being checked. If we want, we can stop the control process after the first error thanks to this PR. Since there is no user_id in this example, we can ensure that the post_id and comment_id fields are not validation.

```
$rules = [
    "user_id"    => ["required", "exists:users"],
    "post_id"    => ["required", "exists:posts"],
    "comment_id" => ["required", "exists:comments"],
];
```


**Use of:**
We can define it by creating a property named $failOnFirstError inside our FormRequest class.
Sample:
```
class FooBarStoreRequest extends FormRequest
{
    protected $failOnFirstError = true;

    /**
     * Determine if the user is authorized to make this request.
     *
     * @return bool
     */
    public function authorize()
    {
        return true;
    }

    /**
     * Get the validation rules that apply to the request.
     *
     * @return array
     */
    public function rules()
    {
        return [
            "foo" => ["required", "string"],
            "baz" => ["required"],
            "age" => ["required", "min:31"]
        ];
    }
}
```


By default, this feature is turned off and its use is optional.


Test processes were added to ``tests/Validation/ValidationValidatorTest.php`` and passed the test successfully.

yusufonur committed:
Added failOnFirstError property

yusufonur committed:
Added failOnFirstError property

yusufonur committed:
Added failOnFirstError property

yusufonur committed:
Added failOnFirstError property

yusufonur committed:
Added failOnFirstError property

yaayes commented:
I think there is a `bail` validation rule for this purpose

yusufonur commented:
> I think there is a `bail` validation rule for this purpose
> 
> https://laravel.com/docs/8.x/validation#rule-bail

Bail works for rules defined for a field.

The property defined in this PR ensures that if one field fails, other fields are not validated.  Can be used with Bail.

GrahamCampbell commented:
Please don't open new PRs every time you make a change.

yaayes commented:
Uhh sorry I misunderstood your PR.

yusufonur commented:
> Please don't open new PRs every time you make a change.

This is the first time I am sending PR, I was a little inexperienced.  I'll pay attention after that.

morloderex commented on src/Illuminate/Foundation/Http/FormRequest.php:
```diff
@@ -65,6 +65,22 @@ class FormRequest extends Request implements ValidatesWhenResolved
+    {
+        parent::__construct($query, $request, $attributes, $cookies, $files, $server, $content);
+
+        if (! isset($this->failOnFirstError)) {
```
```suggestion
        if (! property_exists($this->failOnFirstError)) {
```

yusufonur committed:
Update src/Illuminate/Foundation/Http/FormRequest.php

Co-authored-by: Michael Lundbøl <michael.lundboel@gmail.com>

yusufonur committed:
Update FormRequest.php

property_exists is not available. Because we want the FormRequest class to be overridden.

dkulyk commented on src/Illuminate/Foundation/Http/FormRequest.php:
```diff
@@ -65,6 +65,22 @@ class FormRequest extends Request implements ValidatesWhenResolved
+    {
+        parent::__construct($query, $request, $attributes, $cookies, $files, $server, $content);
+
+        if (! property_exists($this->failOnFirstError)) {
```
[property_exists docs](https://www.php.net/manual/en/function.property-exists.php)
But $this  already have `failOnFirstError` property. 

Why not just `protected $failOnFirstError = false;`?

yusufonur replied:
You are right. This questioning was unnecessary. Thanks.

yusufonur committed:
Update FormRequest.php

taylorotwell commented:
I was able to add support for this with no breaking changes:

```php
$validator = Validator::make(...)->stopOnFirstFailure();

// From form request...

protected $stopOnFirstFailure = true;
```
